src/coprocessor/dag/executor/topn.rs

```diff
@@ -109,15 +112,22 @@ impl TopNExecutor {
     }
 }
 
-impl Executor for TopNExecutor {
+impl<C: ExecSummaryCollector> Executor for TopNExecutor<C> {
     fn next(&mut self) -> Result<Option<Row>> {
+        let timer = self.summary_collector.on_start_iterate();
```

Comment:
No need to add [`next_impl`](https://github.com/tikv/tikv/pull/4699#issuecomment-492751938) for this? The `self.fetch_all()?` on line 119 will quit, for instance.

Reply:
addressed